How to implement topology awareness hint

Manyamteja opened the pull request:
*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Manyamteja committed:
how to implement topology awareness hint

how to implement topology awareness hint

berry2012 reviewed (changes requested):
Ensure each container image has a pinned (tag) version and latest is not used
```
      containers:
      - image: nginx:latest
```

Manyamteja committed:
Update demo-nginx-deployment.yaml

Changed the Image pointing to public ECR gallery

Manyamteja committed:
Update demo-nginx-deployment.yaml

Manyamteja committed:
Update demo-nginx-deployment.yaml

Added the Pod Security contexts

Manyamteja committed:
Merge branch 'aws-samples:main' into main

Manyamteja committed:
Update readme.md

Manyamteja committed:
Update readme.md

Manyamteja committed:
Merge branch 'aws-samples:main' into main

Manyamteja committed:
Update demo-namespace.yaml

Added PSS at the namespace

Manyamteja committed:
Changing the image to busybox 

Using busybox image instead of Nginx

Manyamteja committed:
Updated the deprecated annotation 

Changed the annotation to service.kubernetes.io/topology-mode: auto

Manyamteja committed:
Updated readme.md to reflect the changes as image is updated to busybox

Manyamteja committed:
Update readme.md

Corrected the image references from Nginx to Busybox
Added a point in Caveat section to mention about Fargate

Manyamteja committed:
Rename service-demo-nginx.yaml to service-demo-busybox.yaml

Renamed the file name
